pkg/operator/secretannotator/aws/reconciler.go

```diff
@@ -78,37 +115,61 @@ type ReconcileCloudCredSecret struct {
 	AWSClientBuilder func(accessKeyID, secretAccessKey []byte, c client.Client) (ccaws.Client, error)
 }
 
-// Reconcile will annotate the cloud cred secret to indicate the capabilities of the cred's capabilities:
+// Reconcile will typically annotate the cloud cred secret to indicate the capabilities of the cred's capabilities:
 // 1) 'mint' for indicating that the creds can be used to create new sub-creds
 // 2) 'passthrough' for indicating that the creds are capable enough for other components to reuse the creds as-is
 // 3) 'insufficient' for indicating that the creds are not usable for the cluster
+// In the event that the operator config resource has specified a mode to operate under (mint/passthrough)
+// then skip trying to determine the capabilities, and just annotate the secret.
 // +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;update
 func (r *ReconcileCloudCredSecret) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	start := time.Now()
 
-	r.Logger.Info("validating cloud cred secret")
-
-	operatorIsDisabled, err := utils.IsOperatorDisabled(r.Client, r.Logger)
+	mode, disabled, conflict, err := utils.GetOperatorConfiguration(r.Client, r.Logger)
```

Comment:
I think this should be done after the defer to record metrics. We would want to record the metrics for the reconcile even if the mode is manual, there is a conflict, or there is an error determining the mode.

Reply:
done